eth/protocols/eth/peer.go

```diff
@@ -465,3 +424,32 @@ func (p *Peer) RequestTxs(hashes []common.Hash) error {
 		GetPooledTransactionsPacket: hashes,
 	})
 }
+
+type set struct {
```

Comment:
nit: I don't like the word `set`, because it doesn't convey that it's a "leaky" set. Perhaps "constrainedSet" or "leakySet" or something.

Reply:
Indeed, it was a placeholder and I forgot to find something better. I will use leakySet.